components/homme/src/prim_movie_mod.F90

```diff
@@ -219,10 +224,18 @@ subroutine prim_movie_init(elem, par, hvcoord,tl)
     call nf_global_attribute(ncdf, 'np', np)
     call nf_global_attribute(ncdf, 'ne', ne)
 
+  if (geometry=="sphere") then
```

Comment:
not sure how slow string comparison can be, could this be replaced with boolean quantities, like sphgeometry = (geometry == "sphere")?

Reply:
it definitely could be! I am not sure if the performance matters much here though, since these comparisons happen only in initialization where I don't think the difference between boolean and string comparison should matter.

Reply:
string comparisons should be avoided inside performance critical loops.  But at this level in the code, I think they are easier to read and  I vote to keep it.